src/coreclr/jit/morph.cpp

```diff
@@ -11198,11 +11210,9 @@ GenTree* Compiler::fgMorphCopyBlock(GenTree* tree)
         }
 #endif // TARGET_ARM
 
-        // Don't use field by field assignment if the src is a call
-        //
-        // TODO: Document why do we have this restriction?
-        //       Maybe it isn't needed, or maybe it is only needed
-        //       for calls that return multiple registers
+        // Don't use field by field assignment if the src is a call,
+        // lowering will handle it without spilling the call result into memory
+        // to access the individual fields.
```

Comment:
Two questions:

1. What is the behavior when assigning a call result to  a promoted struct  that has a single field.

2. How does lower patch things up?   
    Don't we create a copy block here, which makes the dest become address taken?
    How can lower fix that?

Reply:
> 1. What is the behavior when assigning a call result to a promoted struct that has a single field.

This is a special case covered by `CanBeReplacedWithItsField`, it checks exactly for promoted struct with 1 field without holes, in this case the dst is not address taken. it happens in `LowerStoreLocCommon`:
https://github.com/dotnet/runtime/blob/54d2b3569172e8ac0873bde180d39511ced66ce8/src/coreclr/jit/lower.cpp#L3119

> 2. How does lower patch things up?

for the case of `ASG(LCL_VAR, call)` it would come as `STORE_LCL_VAR(call)` to `LowerStoreSingleRegCallStruct` and it will generate `STORE_IND` or `STORE_BLK` . Store_blk is used when the struct has the size of 3, 5, 6, 7 bytes. The details are in  `LowerCallStruct`.
https://github.com/dotnet/runtime/blob/54d2b3569172e8ac0873bde180d39511ced66ce8/src/coreclr/jit/lower.cpp#L3543